osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/mbus/ScanMbusChannelsCommandExecutor.java

```diff
@@ -147,15 +147,19 @@ private String determineIdentificationNumber(
     final Long identification =
         this.dlmsHelper.readLong(getResult, "Identification number on channel " + channel);
 
-    final IdentificationNumber identificationNumber;
+    try {
+      final IdentificationNumber identificationNumber;
 
-    if (this.identificationNumberStoredAsBcdOnDevice(mBusClientSetupVersion)) {
-      identificationNumber = IdentificationNumber.fromBcdRepresentationAsLong(identification);
-    } else {
-      identificationNumber = IdentificationNumber.fromNumericalRepresentation(identification);
-    }
+      if (this.identificationNumberStoredAsBcdOnDevice(mBusClientSetupVersion)) {
+        identificationNumber = IdentificationNumber.fromBcdRepresentationAsLong(identification);
+      } else {
+        identificationNumber = IdentificationNumber.fromNumericalRepresentation(identification);
+      }
 
-    return identificationNumber.getTextualRepresentation();
+      return identificationNumber.getTextualRepresentation();
+    } catch (final IllegalArgumentException e) {
+      return getResult.getResultData().toString() + " (Cannot not be correctly interpreted)";
```

Comment:
Zou ResultData null kunnen zijn?

Reply:
In this context the IllegalArgumentException is not thrown when ResultData is null but it would be more safe to do a null check